kahuna/public/js/search/structured-query/structured-query.js

```diff
@@ -67,6 +68,22 @@ grStructuredQuery.directive('grStructuredQuery', ['subscribe$', function(subscri
 
             subscribe$(scope, ctrl.newQuery$, query => {
                 ngModelCtrl.$setViewValue(query);
+                const structuredQuery = structureQuery(query);
+                structuredQuery.forEach(queryComponent => {
+                    // e.g. filter or search:
+                    // search > {type: 'text', value: 'my search'}
+                    // filter > {type: 'filter', filterType: 'inclusion', key: 'is', value: 'cool'}
+                    const { type } = queryComponent;
+                    const formattedType = (type) => {
+                        if (type === 'text') { return 'GRID_SEARCH'; }
+                        if (type === 'filter') { return 'GRID_FILTER'; }
+                        return `GRID_${type.toUpperCase()}`;
+                    };
+                    if (queryComponent.value){
+                        // In case search is empty, as with a search containing only filters
```

Comment:
if i add a search, and then delete it, grid sends an event with value: "undefined" (the string undefined, not the value) - I think this is a bug in structureQuery

Reply:
https://github.com/guardian/grid/blob/main/kahuna/public/js/search/structured-query/syntax.js#L36-L40 `'' + undefined === 'undefined'` 🙃

Reply:
Made some changes that may have sorted this - will check in the morning